### PR TITLE
*: Sort dependencies in Cargo.toml in order automatically

### LINKED ISSUE #12444: Sort dependencies in Cargo.toml in order
## Development Task
Sort dependencies in Cargo.toml in order, it's all in the muddle now.

### PR DESCRIPTION
**Pull request by Connor1996**
Signed-off-by: Connor1996 <zbk602423539@gmail.com>

### What is changed and how it works?

Issue Number: Close #12444 

What's Changed: Utilize https://github.com/DevinR528/cargo-sort to sort the dependencies.

```commit-message
Sort dependencies in Cargo.toml in order automatically
```

### Check List 

Tests 

- No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -271,10 +271,11 @@ unset-override:
 
 pre-format: unset-override
 	@rustup component add rustfmt
+	@cargo install -q cargo-sort 
 
 format: pre-format
-	@cargo fmt -- --check >/dev/null || \
-	cargo fmt
+	@cargo fmt -- --check >/dev/null || cargo fmt
```
Comment (tabokie) [inline Makefile:277]: The first check seems useless.

**Review by tabokie — APPROVED** [review]

**Comment by BusyJay** [comment]
What's the point of sorting?

File: Makefile (hunk 1 above)
Comment (Connor1996) [reply Makefile:277]: The first check is to do the actual format, and the second is to print out the error if there is something wrong with the `cargo fmt`

**Comment by Connor1996** [comment]
> What's the point of sorting?

neater than unordered

File: Makefile (hunk 1 above)
Comment (tabokie) [reply Makefile:277]: No, the check won't do format. It prints unformatted code to stdout and leave it at that.

I think this line first scans the code in read-only mode, and then do the actual format if needed. But it's not like "format" has a big overhead. I don't see why we need to check in read-only mode first.

Comment (Connor1996) [reply Makefile]: I see

**Comment by BusyJay** [comment]
> neater than unordered

They have been sorted in the [past](https://github.com/tikv/tikv/pull/6079), and now they need to be sorted again. I don't think we need to bother change it. Or there should be tools to guarantee its order automatically.

**Comment by Connor1996** [comment]
> > neater than unordered
> 
> They have been sorted in the [past](https://github.com/tikv/tikv/pull/6079), and now they need to be sorted again. I don't think we need to bother change it. Or there should be tools to guarantee its order automatically.

@BusyJay Please check the change of Makefile, it's guaranteed by cargo-sort when running `make format`

**Review by BornChanger — COMMENTED** [review]
Just curious, if some IDE support automatic sorting?

**Review by innerr — APPROVED** [review]

**Comment by BornChanger** [comment]
> make format

Saw your last comment. make format can sort automatically.

**Review by BornChanger — APPROVED** [review]

**Pull request merged**